covid-vaccine-faq-bot/functions/es-dialogflow-detect-intent.js

```diff
@@ -0,0 +1,42 @@
+// Imports the Google Cloud API library
+const {SessionsClient} = require('@google-cloud/dialogflow');
+ 
+exports.handler = async function (context, event, callback) {
+ 
+ let query = event.utterance;
+ 
+const client = new SessionsClient({
+  keyFilename: Runtime.getAssets()[context.GOOGLE_APPLICATION_CREDENTIALS].path
+});
+ 
+ // dialogflow keeps conversation state organized by session ids.  in order to have back and forth passes with dialogflow, we need to maintain a consistent
+ // session id throughout the dialog.  if one doesn't exist, generate it.  if it does, use what we have.
+ if (!event.dialogflow_session_id){
+   event.dialogflow_session_id = Math.random().toString(36).substring(7);
```

Comment:
lets use constants to define what these magic numbers are

Reply:
is there also a a reference as to how these session id's should be generated?

Reply:
Alos this thread points out the flaws in generating a id this way.. have we accounted for this?

https://stackoverflow.com/questions/1349404/generate-random-string-characters-in-javascript

Reply:
@jme783 @sudheerchekka to take care of these comments

Reply:
Replaced Math with Crypto module as Crypto generates cryptographically strong random values @anishshandilya @iaggarwal27 @jme783

Reply:
Thank you @sudheercheckka!